test/e2e/tests/test_notebook_instance_lifecycle.py

```diff
@@ -0,0 +1,133 @@
+"""Integration tests for the Notebook Lifecycle configuration
+"""
+
+import pytest
+import logging
+import botocore
+import datetime
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import (
+    service_marker,
+    wait_for_status,
+    create_sagemaker_resource,
+    sagemaker_client,
+)
+
+from e2e.bootstrap_resources import get_bootstrap_resources
+import random
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from time import sleep
+
+
+@pytest.fixture(scope="function")
+def notebook_instance_lifecycleConfig():
+    notebook_instance_lfc_name = random_suffix_name("notebookinstancelfc", 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural="notebookinstancelifecycleconfigs",
+        resource_name=notebook_instance_lfc_name,
+        spec_file="notebook_instance_lifecycle_config",
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, 10, 5)
+        assert deleted
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+@pytest.mark.canary
+class TestNotebookInstanceLifecycleConfig:
+    def wait_until_update(self, reference, current_time):
+        for i in range(10):
+            resource = k8s.get_resource(reference)
+            assert resource is not None
+            assert "lastModifiedTime" in resource["status"]
+            lastModifiedTime = resource["status"]["lastModifiedTime"]
+            d = datetime.datetime.strptime(lastModifiedTime, "%Y-%m-%dT%H:%M:%SZ")
+            if d > current_time:
+                return
+            sleep(1)
+        return
+
+    def test_CreateUpdateDeleteNotebookLifecycleConfig(
+        self, notebook_instance_lifecycleConfig
+    ):
+        (reference, resource, spec) = notebook_instance_lifecycleConfig
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+        current_time = datetime.datetime.today()
+        sleep(5)  # Done to avoid flakiness since update happens instantaneously.
```

Comment:
? update is at line 113

Reply:
Yea but it takes 0.211 seconds to get there, and the smallest unit of last modified time is in seconds eg 2021-08-12 04:07:21 where 21 is seconds.

Reply:
don't use `current_time` 

instead just store the `lastModifiedTime` and see if it changes

Reply:
used lasModifiedtime and that seems to work well